packages/kiota_abstractions/lib/src/abstract_query_parameters.dart

```diff
@@ -0,0 +1,6 @@
+part of '../kiota_abstractions.dart';
+
+/// Type definition for query parameters.
+abstract class AbstractQueryParameters{
+  Map<String, dynamic> getQueryParameters();
```

Comment:
Maybe we should rename this method to `toMap` as the parameters are already represented by the object itself and this method returns a `Map` representation of them